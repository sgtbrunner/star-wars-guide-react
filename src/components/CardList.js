import React from 'react';
import Card from './Card';

const CardList = ({ filteredCharacters, openModal }) => {
  if (filteredCharacters.length) {
    return (
      <div id="cardlist">
        {filteredCharacters.map((character) => {
          return (
            <Card
              name={character.name}
              index={character.id}
              key={character.id}
              openModal={openModal}
            />
          );
        })}
      </div>
    );
  }
  return (
    <div className="pa2 ma3 custom-field tc">
      <p1 className="custom-font animate-flicker tc">No characters found!</p1>
    </div>
  );
};

export default CardList;
