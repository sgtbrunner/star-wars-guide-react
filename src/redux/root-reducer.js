import { combineReducers } from 'redux';

import charactersReducer from './characters/characters.reducer';
import dialogReducer from './dialog/dialog.reducer';
import filmsReducer from './films/films.reducer';
import planetsReducer from './planets/planets.reducer';
import speciesReducer from './species/species.reducer';

export default combineReducers({
  characters: charactersReducer,
  films: filmsReducer,
  planets: planetsReducer,
  species: speciesReducer,
  dialog: dialogReducer,
});
