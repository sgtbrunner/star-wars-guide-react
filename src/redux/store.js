import { createStore, applyMiddleware } from 'redux';
import { persistStore } from 'redux-persist';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';

import rootReducer from './root-reducer';

const middlewares = [thunk];

export const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(...middlewares)));

export const persistor = persistStore(store);

export default { store, persistor };
