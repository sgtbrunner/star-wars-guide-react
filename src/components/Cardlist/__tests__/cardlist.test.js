import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';

import CardList from '../cardlist.component';

const BUTTON = 'button';

const loadedCharactersA = [
  {
    id: 0,
    name: 'Anakin Skywalker',
  },
  {
    id: 1,
    name: 'Obi-Wan Kenoby',
  },
  {
    id: 2,
    name: 'Yoda',
  },
];

const loadedCharactersB = [
  ...loadedCharactersA,
  {
    id: 3,
    name: 'Darth Vader',
  },
  {
    id: 4,
    name: 'Palpatine',
  },
];

const store = createStore(() => {});
const mountCardListComponent = (characters = []) =>
  render(
    <Provider store={store}>
      <CardList characters={characters} />
    </Provider>
  );

describe('CardList component', () => {
  it('should match snapshot when characters are NOT supplied to component', () => {
    const wrapper = mountCardListComponent();

    expect(wrapper).toMatchSnapshot();
  });

  it(`should render 'No characters found' alert if characters are NOT supplied to
    component`, () => {
    mountCardListComponent();

    const cards = screen.queryAllByRole('button');
    expect(cards.length).toBe(0);

    const alert = screen.getByRole('alert');
    expect(alert).toBeInTheDocument();
    expect(alert).toHaveTextContent(/no characters found!/i);
  });

  it('should match snapshot when characters are supplied to component', () => {
    const wrapper = mountCardListComponent(loadedCharactersA);

    expect(wrapper).toMatchSnapshot();
  });

  it(`should render three (3) cards if three (3) characters are supplied to
    component`, () => {
    mountCardListComponent(loadedCharactersA);

    const cards = screen.getAllByRole(BUTTON);
    expect(cards).toHaveLength(3);
  });

  it(`rendered cards should match character names when three (3) characters are supplied 
    to component`, () => {
    mountCardListComponent(loadedCharactersA);

    const cards = screen.getAllByRole(BUTTON);
    cards.forEach((card, index) => {
      expect(card).toHaveTextContent(loadedCharactersA[index].name);
    });
  });

  it(`should render five (5) cards if five (5) characters are supplied to
  component`, () => {
    mountCardListComponent(loadedCharactersB);

    const cards = screen.getAllByRole(BUTTON);
    expect(cards).toHaveLength(5);
  });

  it(`rendered cards should match character names when five (5) characters are supplied
    to component`, () => {
    mountCardListComponent(loadedCharactersB);

    const cards = screen.getAllByRole(BUTTON);
    cards.forEach((card, index) => {
      expect(card).toHaveTextContent(loadedCharactersB[index].name);
    });
  });
});
