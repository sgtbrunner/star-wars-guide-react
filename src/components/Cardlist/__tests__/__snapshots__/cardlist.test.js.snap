// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList component should match snapshot when characters are NOT supplied to component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pa2 ma3 tc white-backdrop"
        role="alert"
      >
        <p
          class="alert-text animate-flicker"
        >
          No characters found!
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pa2 ma3 tc white-backdrop"
      role="alert"
    >
      <p
        class="alert-text animate-flicker"
      >
        No characters found!
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CardList component should match snapshot when characters are supplied to component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cardlist"
      >
        <button
          class="card pointer"
          id="0"
          type="button"
        >
          <img
            alt="character-portrait"
            id="0"
            src="https://starwars-visualguide.com/assets/img/characters/1.jpg"
          />
          <div
            class="name"
            id="0"
          >
            Anakin Skywalker
          </div>
        </button>
        <button
          class="card pointer"
          id="1"
          type="button"
        >
          <img
            alt="character-portrait"
            id="1"
            src="https://starwars-visualguide.com/assets/img/characters/2.jpg"
          />
          <div
            class="name"
            id="1"
          >
            Obi-Wan Kenoby
          </div>
        </button>
        <button
          class="card pointer"
          id="2"
          type="button"
        >
          <img
            alt="character-portrait"
            id="2"
            src="https://starwars-visualguide.com/assets/img/characters/3.jpg"
          />
          <div
            class="name"
            id="2"
          >
            Yoda
          </div>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cardlist"
    >
      <button
        class="card pointer"
        id="0"
        type="button"
      >
        <img
          alt="character-portrait"
          id="0"
          src="https://starwars-visualguide.com/assets/img/characters/1.jpg"
        />
        <div
          class="name"
          id="0"
        >
          Anakin Skywalker
        </div>
      </button>
      <button
        class="card pointer"
        id="1"
        type="button"
      >
        <img
          alt="character-portrait"
          id="1"
          src="https://starwars-visualguide.com/assets/img/characters/2.jpg"
        />
        <div
          class="name"
          id="1"
        >
          Obi-Wan Kenoby
        </div>
      </button>
      <button
        class="card pointer"
        id="2"
        type="button"
      >
        <img
          alt="character-portrait"
          id="2"
          src="https://starwars-visualguide.com/assets/img/characters/3.jpg"
        />
        <div
          class="name"
          id="2"
        >
          Yoda
        </div>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
